src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -3809,7 +3888,8 @@ internal static bool AreRefsCompatibleForMethodConversion(RefKind candidateMetho
             MethodSymbol method;
             EffectiveParameters constructedEffectiveParameters;
             bool hasTypeArgumentsInferredFromFunctionType = false;
-            if (member.Kind == SymbolKind.Method && (method = (MethodSymbol)(Symbol)member).Arity > 0)
+            if ((options & Options.InferringUniqueMethodGroupSignature) == 0 &&
```

Comment:
It is not clear what is happening with generic methods when we get here. It looks like sometimes we are getting called with constructed-from symbols (from ```GetUniqueSignatureFromMethodGroup_CSharp10```) and sometimes with constructed symbols (from ```GetUniqueSignatureFromMethodGroup```). Are we certain that, when we manage to filter something out, we are getting the same symbols for the candidates that passed the filter?

Reply:
> Are we certain that, when we manage to filter something out, we are getting the same symbols for the candidates that passed the filter?

I think the only place where overload resolution could replace the member with its constructed form is inside this `if`, but we won't get into it because of the check `(options & Options.InferringUniqueMethodGroupSignature) == 0`. We are also passing no arguments and no type arguments to overload resolution, so it's not be possible for overload resolution to replace the member with its constructed form.